pkg/apis/kubernetes/v1/crud_test.go

```diff
@@ -0,0 +1,390 @@
+	testutils "go.anx.io/go-anxcloud/pkg/utils/test"
+)
+
+const (
+	mockKubernetesVersion = "1.23.6"
+	nonExistingIdentifier = "non-existing-identifier"
+)
+
+var _ = Describe("CRUD", Ordered, func() {
+	var (
+		a   api.API
+		srv *ghttp.Server
+
+		clusterIdentifier  string
+		nodePoolIdentifier string
+
+		clusterName  = "go-anxcloud-" + testutils.RandomHostname()
+		nodePoolName = "go-anxcloud-" + testutils.RandomHostname()
+		location     = corev1.Location{Identifier: "52b5f6b2fd3a4a7eaaedf1a7c019e9ea"} // ANX04
+	)
+
+	mockStateOK := map[string]interface{}{"id": "0"}
+	mockStateError := map[string]interface{}{"id": "1"}
+	mockStatePending := map[string]interface{}{"id": "2"}
+
+	BeforeEach(func() {
+		srv = ghttp.NewServer()
+
+		var err error
+
+		if isIntegrationTest {
+			a, err = api.NewAPI(api.WithClientOptions(client.AuthFromEnv(false)))
+			Expect(err).ToNot(HaveOccurred())
+		} else {
+			a, err = api.NewAPI(api.WithClientOptions(
+				client.BaseURL(srv.URL()),
+				client.IgnoreMissingToken(),
+			))
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	Context("Cluster object", Ordered, func() {
+		Context("Create operation", Ordered, func() {
+			It("Create new cluster", func() {
+				appendCreateClusterHandler(srv,
+					map[string]interface{}{"name": clusterName, "location": location.Identifier},
+					mockedClusterResponse(Cluster{Identifier: "mocked-cluster-identifier", Name: clusterName}),
+				)
+
+				cluster := Cluster{Name: clusterName, Location: location}
+
+				err := a.Create(context.TODO(), &cluster)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(cluster.Identifier).NotTo(BeEmpty())
+				clusterIdentifier = cluster.Identifier
+				Expect(cluster.StatePending()).To(BeTrue())
+				Expect(pointer.BoolVal(cluster.NeedsServiceVMs)).To(BeTrue()) // Default if not set on Create
+			})
+		})
+
+		Context("AwaitCompletion", Ordered, func() {
+			It("can wait until state is ready", func() {
+				appendGetClusterHandler(srv, clusterIdentifier, http.StatusOK, map[string]interface{}{"state": mockStatePending})
+				appendGetClusterHandler(srv, clusterIdentifier, http.StatusOK, map[string]interface{}{"state": mockStatePending})
+				appendGetClusterHandler(srv, clusterIdentifier, http.StatusOK, map[string]interface{}{"state": mockStateOK})
+
+				cluster := Cluster{Identifier: clusterIdentifier}
+				err := cluster.AwaitCompletion(context.TODO(), a)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(cluster.StateOK()).To(BeTrue())
+
+				if !isIntegrationTest {
+					Expect(srv.ReceivedRequests()).To(HaveLen(3))
+				}
+			})
+
+			It("returns an error when updating the state fails", func() {
+				cluster := Cluster{}
+				err := cluster.AwaitCompletion(context.TODO(), a)
+				Expect(err).To(MatchError(types.ErrUnidentifiedObject))
+			})
+
+			Context("mock test", Ordered, func() {
+				BeforeEach(func() {
+					if isIntegrationTest {
+						Skip("Can't be tested with integration tests")
+					}
```

Comment:
sure it can, just not make it use the real client?

Reply:
I'm a little confused :D
We have a separate run where `isIntegrationTest` is false and in this case we are testing with the mock client. Can you please elaborate?

Reply:
Our tests are two modes: unit tests only and all tests (including integration tests) - and I think this clears up what I mean?